CODE_REVIEW_GUIDELINES.md

```diff
@@ -0,0 +1,48 @@
+# Overall:
+
+- Does the contribution referece an existing Issue?
+- Are the requirements well defined?
+
+
+#  Specification:
+
+- Are the requirements for the change well specified? Where are they documented? Bugfixes need a clear specification of the bug's cause and fix. (Small PRs might specify in a github ticket. Large changes may require stand-alone docs, or ZIPs.)
+- Is there a documented test plan, which describes how to manually verify the change works on testnet prior to a release?
+
+# User Documentation:
+
+- What is the "changelog" entry for this change? (All changes should include this.)
+- Are there any changes which require updates to user-facing documentation? If so, does the new documentation make sense? (documentation should be placed in [docs/](/docs) folder of the repo)
+
+# Testing:
+
+- For non-minor PRs (up to the code reviewers and PR creator if this is needed), we require authors to perform a thorough self-review and self-test of the resulting code base, including use cases that might not be visible affected by the introduced changes. Reviewers are not expected to run the changes locally, but are definitely encouraged to do so at their best judgement. 
+- When introducing modifications that affect the UI, screenshots might be provided in a BEFORE/AFTER fashion to speed up UI/UX requirement validation. 
+- Are there new tests that check all of the requirements? If it's a bugfix does it include new tests testing the bug triggering condition? (Do they fail before the fix and pass after the fix?)
+- Do tests include edge cases, error conditions, and "negative case" tests to ensure the software is robust? Example: a function for verifying transaction signatures should include a bunch of tests for invalid signature cases.
+- Do tests include all "logical test coverage" in addition to requirements-focused tests? Logical test coverage verifies behavior of the code that isn't obvious or implied by the requirements themselves.
+- Have all of the automated tests been executed by CI? Read over the CI report to verify this. (Note: our CI system may not currently provide test results prior to review. We should change this! In the meantime, the reviewer should run the tests.)
+- Is it clear that the test plan does actually test the new change? Does the test plan include any edge case / negative case / error conditions?
+
+# Code Review:
+- Does the code structure make sense? Does it follow existing conventions / frameworks of existing code, or is it introducing new abstractions / structure?
+- Is the code style consistent with the [Swift Style Guide](SWIFTLINT.md)?
+- Does every change make sense? Make sure to ask questions, even or especially "basic" coding questions ("what does this mean in Swift?") and domain-newbie questions ("what is the priority system, and why does this line change the priority?").
+
+In summary, here's a checklist that summarizes what reviewers should be looking for when doing a Code Review [1]
```

Comment:
Is this an ordered or unordered list?

Would an ordered list help in terms of clarifying conflicting priorities?

Reply:
I'm having a hard time trying to see any conflicting priorities here since this list is not overly concrete.  "well-designed" and "functionality is good" or "more complex than it needs to be" are all worded such that they aren't drawing hard lines that could cause conflicts.  Did you have anything specific in mind @ccjernigan ?

Reply:
Maybe I'm falling into the trap of `The developer isn’t implementing things they might need in the future but don’t know they need now or aren't using now.`

I imagine at some point we'll add security and performance to this list, both of which often have tradeoffs to consider.

@adamstener is right that the current items are phrased in a way that avoids conflict nicely!

Reply:
I imagine that this particular rule just needs an explanation to be broken is all.  If something needs to be implemented in an incomplete way, there should be a valid reason, and it probably won't conflict with anything else.